drf_extra_fields/fields.py

```diff
@@ -70,7 +72,9 @@ def to_internal_value(self, base64_data):
             if file_extension not in self.ALLOWED_TYPES:
                 raise ValidationError(self.INVALID_TYPE_MESSAGE)
             complete_file_name = file_name + "." + file_extension
-            data = ContentFile(decoded_file, name=complete_file_name)
+            data = SimpleUploadedFile(name=complete_file_name, content=decoded_file,
+                                      content_type=file_mime_type
+                                      )
```

Comment:
As far as i can see we don't need to calculate content_type. Super of `to_internal_value` use's django's image field to create python representation of the file object.  [Here](https://github.com/django/django/blob/master/django/forms/fields.py#L637) you can see the related code block adds `content_type` property to the file object.

Am i missing something?

Reply:
Yes exactly, but thats irrelevant :) 

The idea is, when you use ContentFile then you are ignoring the header part of base64 data (the ```data:image/png;base64,...``` part) - if I'm not missing anything - so if someone wants to do his/her own dispatch by looking at that "header" then that info is completely gone when you use ContentFile. I think we can agree that header is valuable ? no ?

When you omit that header, then the consequence is, you have to go throught the content itself to determine what has been uploaded - which makes sense if you are uploading images -  or try to understand from the file extension, which in some cases doesn't really work, but thats irrelevant again. 

I think it should be user's responsibility to use that "header" part or not, library should preserve as much as info possible (about the sent content) so the user simply has more possibilities.

The summary is, I think its just ignoring information (the header) which it shouldn't. Let me know what you think ?

Reply:
For the image field case: Whatever you provide as `content_type` to SimpleUploadedFile it will be overrode by Django's ImageField.to_python() method. So you will lose the assigned content_type.

While i was thinking i ignore the fact that this is not for images only. For other file types i agree that it will be useful to know provided file content type. But anyone can change the provided header. Do you think this will cause any issues? For example a user can send a word file and provide application/pdf as header. (Well i don't know who would do that :))

Reply:
Yes, simply if the user wants to keep that field, he shouldn't use ImageField :) 

Exactly. Well also anyone can upload any file too. Depends on the use case I think, if you have a very limited set of users with a spesific clients - like a CMS etc - which is not an issue at all (our use case is very similar to that). Since you literally know the users it might not be an issue at all but if its a public api, then it would make more sense to not to rely on that header and check file type with magick maybe - but I think that should be the choice of the user of the library - because there can be many different use cases.

Reply:
I totally agree. 

To make it more clear that the given content type is retrieved from the user not the file itself should we use another attribute name other than `content_type`? (my suggestion would be provided_content_type). That way we make sure that the user of the library will know what it's.

What do you think?